pkg/liquidity-mining/contracts/gauges/L2GaugeCheckpointer.sol

```diff
@@ -148,9 +135,28 @@ contract L2GaugeCheckpointer is IL2GaugeCheckpointer, ReentrancyGuard {
         Address.sendValue(msg.sender, address(this).balance);
     }
 
-    /**
-     * @dev See `IL2GaugeCheckpointer#getTotalBridgeCost`.
-     */
+    /// @inheritdoc IL2GaugeCheckpointer
+    function checkpointGaugesOfTypeAboveRelativeWeight(IGaugeAdder.GaugeType gaugeType, uint256 minRelativeWeight)
+        external
+        payable
+        override
+        withSupportedGaugeType(gaugeType)
+        nonReentrant
+    {
+        // solhint-disable-next-line not-rely-on-time
+        uint256 currentPeriod = _roundDownTimestamp(block.timestamp);
+
+        _checkpointGauges(gaugeType, minRelativeWeight, currentPeriod);
```

Comment:
This function makes the assumption that all gauges will have the same ETH cost. I imagine we have a mix of capped and uncapped arbitrum gauges so this assumption may break if we don't keep the factories in sync.

I think we can query this cost from each gauge similar to how we do it in `getTotalBridgeCost`.

Reply:
Hmm, nice catch!